src/oci-umount.c

```diff
@@ -412,6 +422,119 @@ static int parent_mntid(const char *id, char *path, const struct mount_info *mnt
 	return -1;
 }
 
+static int find_mount(const char*path, struct host_mount_info *mounts_on_host, int nr_mounts) {
+	for (int i=0; i < nr_mounts; i++) {
+		if (strcmp(mounts_on_host[i].path,path) ==0)
+			return 1;
+	}
+	return 0;
+}
+
+static int readconf(const char *id, const char *path, struct host_mount_info *mounts_on_host, int *number) {
+	_cleanup_fclose_ FILE *fp = NULL;
+	_cleanup_free_ char *line = NULL;
+	ssize_t read;
+	char *real_path;
+	size_t len = 0;
+	int nr_umounts = *number;
+	/* Parse oci-umounts.conf file, canonicalize path names and skip
+	 * paths which are not a mountpoint on host */
+	fp = fopen(path, "r");
+	if (fp == NULL) {
+		if (errno == ENOENT) {
+			pr_pwarning("%s: Config file not found: %s", id, path);
+			return 0;
+		}
+		pr_perror("%s: Failed to open config file: %s", id, path);
+		return EXIT_FAILURE;
+	}
+
+	while ((read = getline(&line, &len, fp)) != -1) {
+		bool submounts_only = false;
+
+		/* Get rid of newline character at the end */
+		line[read - 1] ='\0';
+		read--;
+
+		if (iscomment(line))
+			continue;
+
+		if (nr_umounts == MAX_UMOUNTS) {
+			pr_perror("%s: Exceeded maximum number of supported unmounts is %d", id, MAX_UMOUNTS);
+			return EXIT_FAILURE;
+		}
+
+		// If there is a "/*" at the end, only unmount submounts
+		if (read >= 2) {
+			if (line[read - 1] == '*' && line[read - 2] == '/') {
+				submounts_only = true;
+				line[read - 1] = '\0';
+			}
+		}
+
+		real_path = realpath(line, NULL);
+		if (!real_path) {
+			pr_pdebug("%s: Failed to canonicalize path [%s]: %m. Skipping.", id, line);
+			continue;
+		}
+
+		if (find_mount(real_path, mounts_on_host, nr_umounts))
```

Comment:
Can we rename find_mount(). Its too generic and there are too many mounts floating around. How about `is_duplicate_path()` ?

Reply:
Also we probably should drop a debug message that we are ignoring a path because it is a duplicate? It will not be a common situation so will not show up in logs by default. Will show up only if things are configured wrong. For example, when docker also introduces a config file of its own and it will have same entries in /etc/oci-umount.conf as well. That will be a good indicator to update oci-umount package and drop these entries from /etc/oci-umount.conf.

Reply:
Fixed